gdmix-workflow/test/test_workflow_generator.py

```diff
@@ -84,57 +79,33 @@ def test_detext_model_fixed_effect_workflow_generator(self):
-            '--dev_file': 'movieLens/detext/validationData/test_data.tfrecord',
-            '--test_file': 'movieLens/detext/validationData/test_data.tfrecord',
-            '--metadata_file': 'movieLens/per-user/metadata/tensor_metadata.json',
-            '--label_column_name': 'response',
-            '--uid_column_name': 'uid',
-            '--weight_column_name': 'weight',
-            '--feature_bag': 'global',
-            '--keep_checkpoint_max': '1',
-            '--prediction_score_column_name': 'predictionScore',
-            '--out_dir': 'detext-training/global/models'}
+        expected_train_job_param = (
+            Params(uid_column_name='uid', weight_column_name='weight', label_column_name='response', prediction_score_column_name='predictionScore',
+                   prediction_score_per_coordinate_column_name='predictionScorePerCoordinate', action='train', stage='fixed_effect', model_type='detext',
+                   training_score_dir='detext-training/global/training_scores', validation_score_dir='detext-training/global/validation_scores'),
+            DetextArg(feature_names=['label', 'doc_query', 'uid', 'wide_ftrs_sp_idx', 'wide_ftrs_sp_val'], ftr_ext='cnn', num_units=64,
+                      sp_emb_size=1, num_hidden=[0], num_wide=0, num_wide_sp=45, use_deep=True, elem_rescale=True, use_doc_projection=False,
+                      use_usr_projection=False, ltr_loss_fn='pointwise', emb_sim_func=['inner'], num_classes=1, filter_window_sizes=[3], num_filters=50,
+                      explicit_empty=False, use_bert_dropout=False, unit_type='lstm', num_layers=1,
+                      num_residual_layers=0, forget_bias=1.0, rnn_dropout=0.0, bidirectional=False, normalized_lm=False, optimizer='bert_adam',
+                      max_gradient_norm=1.0, learning_rate=0.002, num_train_steps=1000, num_warmup_steps=0, train_batch_size=64,
+                      test_batch_size=64, l1=None, l2=None, train_file='movieLens/detext/trainingData/train_data.tfrecord',
+                      dev_file='movieLens/detext/validationData/test_data.tfrecord', test_file='movieLens/detext/validationData/test_data.tfrecord',
+                      out_dir='detext-training/global/models', max_len=16, min_len=3, vocab_file='movieLens/detext/vocab.txt', we_file=None,
+                      we_trainable=True, PAD='[PAD]', SEP='[SEP]', CLS='[CLS]', UNK='[UNK]', MASK='[MASK]', we_file_for_id_ftr=None,
+                      we_trainable_for_id_ftr=True, PAD_FOR_ID_FTR='[PAD]', UNK_FOR_ID_FTR='[UNK]', random_seed=1234, steps_per_stats=10, num_eval_rounds=None,
+                      steps_per_eval=100, keep_checkpoint_max=1, init_weight=0.1, pmetric='auc', all_metrics=['auc'], score_rescale=None,
+                      add_first_dim_for_query_placeholder=False, add_first_dim_for_usr_placeholder=False, tokenization='punct', resume_training=False,
+                      use_tfr_loss=False, tfr_loss_fn='softmax_loss'))
```

Comment:
Ditto. Feel hard to read parameters.

Reply:
I doubt there is a need to read the parameters: the keys and the types of the values are guarded by the IDE and compiler and values are part of the tests.

Reply:
My concern is that: 
- Developers might use different IDEs. For example, I use vim once in a while whose functionality isn't as rich as IntelliJ 
- Test case shows future developers parameter examples of the method, such long DeTextArg tuple mixed with type of list, string, int, etc is a bit scary.

Reply:
IDE is a helper, and it's also guarded by python compiler and interpreter.
It's much less "scary" as plain strings.

Reply:
okay.